### ISSUE TITLE
Bug: Output gets hidden on stop_and_persist in Jupyter notebooks

### ISSUE THREAD

**Issue by ManrajGrover** [ManrajGrover/halo#59]
## Description
Currently, in jupyter notebooks, the final `stop_and_persist` output is not persisted. I've opened an issue in ipywidget for understanding the issue a little better (https://github.com/jupyter-widgets/ipywidgets/issues/2072).

### System settings

- Operating System: Mac OS 10.13.1
- Terminal in use: NA
- Jupyter version: 4.4.0
- Python version: 2.7.14
- Halo version: 9f9ea376d43f0d43859f7ddeee68e34834fdbf19
- `pip freeze` output: NA

### Error
The output display is not persited on `stop_and_persist` call.
 
### Expected behaviour
The output should be persited on `stop_and_persist` call.

## Steps to recreate

![widget bug](https://user-images.githubusercontent.com/5673050/40014254-ae56e4e8-57cd-11e8-888b-3ab466977862.gif)

## People to notify
@JungWinter @ManrajGrover

**Comment by ManrajGrover** [comment]
@JungWinter Could you look into this as well?

**Comment by ManrajGrover** [comment]
@JungWinter Yes, it seems to resolve the issue. I'm not sure if it breaks anything else.

**Comment by mschmo** [comment]
I'd love to check to check this out for hacktoberfest. @JungWinter thanks for the starting point.

**Comment by mschmo** [comment]
I looked into this a bit today. [This commit](https://github.com/manrajgrover/halo/commit/9b23858d8f78fe0b2d97641452067012821e0cee) caused the bug.

The reason why is `ip.events.register('post_run_cell', clean_up)` fires after every cell execution, which calls `clear()` and clears the last output.

I don't want to simply revert that commit, because it seems like there was a good reason for wanting to clean up after execution. Maybe @manrajgrover you can expand on what issue that commit is fixing? I notice it replaced a previous callback to handle `SIGINT`.
